.azure-pipelines/ultimate-pipeline.yml

```
@@ -1786,6 +1863,114 @@ stages:
         testResultsFiles: profiler/build_data/results/**/*.trx
       condition: succeededOrFailed()
 
+- stage: asan_profiler_tests
+  #address sanitizer tests
+  condition: >
+    and(
+      succeeded(), 
+      eq(variables['isBenchmarksOnlyBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Linux,Windows]
+
+  - job: Linux
+    timeoutInMinutes: 60 #default value
+
+    pool:
+      name: azure-linux-scale-set
+
+    steps:
+    - template: steps/clone-repo.yml
+      parameters:
+        masterCommitId: $(masterCommitId)
+
+    - template: steps/install-latest-dotnet-sdk.yml
+
+    - template: steps/run-in-docker.yml
+      parameters:
+        build: true
+        target: builder
+        command: "Clean RunProfilerAsanTest -Framework net7.0"
```

nit: presumably don't need the Clean here
```suggestion
        command: "RunProfilerAsanTest -Framework net7.0"
```